android/src/main/java/com/baseflow/permissionhandler/PermissionHandlerPlugin.java

```diff
@@ -250,6 +253,21 @@ public void onMethodCall(MethodCall call, Result result) {
 
   @PermissionStatus
   private int checkPermissionStatus(@PermissionGroup int permission) {
+    final Context context = mRegistrar.activity() == null ? mRegistrar.activeContext() : mRegistrar.activity();
+    if (context == null) {
+      Log.d(LOG_TAG, "Unable to detect current Activity or App Context.");
+      return PERMISSION_STATUS_UNKNOWN;
+    }
+
+    if (permission == PERMISSION_GROUP_NOTIFICATION) {
+      NotificationManagerCompat manager = NotificationManagerCompat.from(context);
+      boolean isGranted = manager.areNotificationsEnabled();
+      if (isGranted) {
+        return PERMISSION_STATUS_GRANTED;
+      }
+      return PERMISSION_STATUS_DENIED;
```

Comment:
To keep the `checkPermissionStatus` method  from growing to big, maybe we can put this code block in a separate method. Something like:

```
private int checkNotificationPermissions(Context context) {
      NotificationManagerCompat manager = NotificationManagerCompat.from(context);
      boolean isGranted = manager.areNotificationsEnabled();
      if (isGranted) {
        return PERMISSION_STATUS_GRANTED;
      }
      return PERMISSION_STATUS_DENIED;
}
```

Reply:
I add a `checkNotificationPermissionStatus` method to check the notification permission status.